File: backend/src/main/java/io/tingkai/money/entity/Stock.java
package io.tingkai.money.entity;

import java.time.LocalDateTime;
import java.util.UUID;

import io.tingkai.money.constant.DatabaseConstant;
import io.tingkai.money.enumeration.MarketType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Data
@Table(name = DatabaseConstant.TABLE_STOCK)
public class Stock {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	protected UUID id;
	@Column(unique = true)
	protected String code;
	protected String name;
	@Column(unique = true)
	protected String isinCode;
	protected String currency;
	protected LocalDateTime offeringDate;
	@Enumerated(EnumType.STRING)
	protected MarketType marketType;
	protected String industryType;
	protected String cfiCode;
	protected String symbol;
	protected String description;
}

File: python/src/entity/Stock.py
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.schema import UniqueConstraint
import uuid

from entity.BaseEntity import BaseEntity, db

class Stock(BaseEntity):
    __tablename__ = 'stock'

    id = db.Column(UUID(as_uuid = True), primary_key = True, default = uuid.uuid4)
    code = db.Column(db.String, nullable = False)
    name = db.Column(db.String, nullable = False)
    isin_code = db.Column(db.String, nullable = False)
    currency = db.Column(db.String, nullable = False)
    offering_date = db.Column(db.DateTime, nullable = False)
    market_type = db.Column(db.String, nullable = False)
    industry_type = db.Column(db.String)
    cfi_code = db.Column(db.String, nullable = False)
    symbol = db.Column(db.String)
    description = db.Column(db.String)

    UniqueConstraint('code', name='stock_code_key')
    UniqueConstraint('isin_code', name='stock_isin_code_key')
